src/mrpro/data/_DcfData.py

```diff
@@ -91,14 +91,17 @@ def _dcf_2d3d_voronoi(traj: torch.Tensor) -> torch.Tensor:
             density compensation values (1, k2, k1, k0)
         """
 
+        # Get device of trajectory
+        traj_device = traj.device
+
         # 2D and 3D trajectories supported
         dim = traj.shape[0]
         if dim not in (2, 3):
             raise ValueError(f'Only 2D or 3D trajectories supported, not {dim}D.')
 
         # Calculate dcf only for unique k-space positions
         traj_dim = traj.shape
-        traj = np.round(traj.numpy(), decimals=UNIQUE_ROUNDING_DECIMALS)
+        traj = np.round(traj.cpu().numpy(), decimals=UNIQUE_ROUNDING_DECIMALS)
```

Comment:
Maybe we can use the oppertunity to give the input traj  and the numpy traj different names? 
This would allow mypy to consider them as array / tensor not array|tensor

this would also remove `traj_device = traj.device`

Reply:
I left the input traj as traj because this should be our standard traj tensor.